core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -259,7 +259,7 @@ private Result visitAggregate(Aggregate e, List<Integer> groupKeyList) {
     // "select a, b, sum(x) from ( ... ) group by a, b"
     final Result x = visitChild(0, e.getInput());
     final Builder builder;
-    if (e.getInput() instanceof Project) {
+    if ((e.getInput() instanceof Project) && (groupKeyList.isEmpty() || !x.withGroupBy())) {
```

Comment:
If the child is a pure `Project` , i.e. no grouping at all, why the condition here `groupKeyList.isEmpty()` is necessary ?

Reply:
To be concrete, can we test condition `groupKeyList.isEmpty()` ?

Reply:
Pr has been updated, would you like to review it for me?